CONTRIBUTING.md

```diff
@@ -43,11 +71,88 @@ Adding the following lines to the `Golang` extension configuration file will ena
 },
 ```
 
-### GoLand / IntelliJ
+### GoLand / IntelliJ requirements
 
-In GoLand or IntelliJ, the plugin [Go Linter](https://plugins.jetbrains.com/plugin/12496-go-linter) will be required.
+- Install either the **GoLand** or **IntelliJ**  Integrated Development Environment (IDE) for the Go programming language, plus the [Go Linter](https://plugins.jetbrains.com/plugin/12496-go-linter) plugin.
 
-The plugin can be installed via `Settings` >> `Plugins` >> `Marketplace`, search for `Go Linter` and install it.
+- The plugin can be installed via `Settings` >> `Plugins` >> `Marketplace`, search for `Go Linter` and install it.
 Once installed, make sure that the plugin is using the `.golangci.yml` file from the root directory.
 
-The configuration of `Go Linter` can be found in the `Tools` section of the settings.
+- The configuration of `Go Linter` can be found in the `Tools` section of the settings.
+
+If you are on Windows, you need to install **make** for the above process to complete.
+
+( **NOTE**: When using the make command on Windows, you may receive an `unrecognized command` error for a command that is installed.
+This usually indicates that `PATH` for the binary is not set correctly).
+
+## Submit a Pull Request 🚀
+
+At this point, you should switch back to the `main` branch in your repository, and make sure it is up to date with `main` branch of Keptn:
+
+```bash
+git remote add upstream https://github.com/keptn/lifecycle-toolkit.git
+git checkout main
+git pull upstream main
+```
+
+Then update your feature branch from your local copy of `main` and push it:
+
+```bash
+git checkout feature/123/foo
+git rebase main
+git push --set-upstream origin feature/123/foo
+```
+
+**All PRs must include a commit message with a description of the changes made!**
+
+
+### Commit Types
+
+`Type` can be:
+
+- `feat    `: A new feature
+- `fix     `: A bug fix
+- `build   `: Changes that affect the build system or external dependencies
+- `chore   `: Other changes that don't modify source or test files
+- `ci      `: Changes to our CI configuration files and scripts
+- `docs    `: Documentation only changes
+- `perf    `: A code change that improves performance
+- `refactor`: A code change that neither fixes a bug nor adds a feature
+- `revert  `: Reverts a previous commit
+- `style   `: Changes that do not affect the meaning of the code
+- `test    `: Adding missing tests or correcting existing tests
+
+
+## Auto signoff commit messages
+
+We have a DCO check that runs on every PR to verify that the commit has been signed off.
```

Comment:
I would add also some info that you can sign your commits when you make them using the `-s` flag, e.g., `git commit -sm "my awesome contribution"`

Reply:
Oh, yes! I added the whole PR section now and commit instructions. I think now it's LGTM. @thisthat :)